### PR TITLE
Maintenance: improve error feedback when tasks fail in CI

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

Changed the output of JUnit.XML files when there is an arbitrary code error (not a failure coming from a test)

**From:**
```xml
<testsuites tests="1" errors="1" failures="0" skipped="0" name="e2e-tests - react-vite/default-js" time="0.001">
  <testsuite tests="1" errors="1" failures="0" skipped="0" name="e2e-tests - react-vite/default-js" time="0.001" timestamp="2022-11-09T08:48:00.915Z">
    <testcase name="e2e-tests - react-vite/default-js" classname="react-vite/default-js e2e-tests - react-vite/default-js" time="0.001" assertions="1">
      <error>iDoNotExist is not defined</error>
    </testcase>
  </testsuite>
</testsuites>
```

**To:**
```xml
<testsuites tests="1" errors="0" failures="0" skipped="0" name="e2e-tests - react-vite/default-js" time="0.001">
    <testsuite tests="1" errors="0" failures="0" skipped="0" name="e2e-tests - react-vite/default-js" time="0.001" timestamp="2022-11-09T08:17:43.019Z">
        <testcase name="e2e-tests - react-vite/default-js" classname="react-vite/default-js react-vite/default-js react-vite/default-js e2e-tests - react-vite/default-js" time="0.001" assertions="1">
            <error>
                ReferenceError: iDoNotExist is not defined
                    at Object.run (/Users/yannbraga/open-source/storybook/scripts/tasks/e2e-tests.ts:13:17)
                    at runTask (/Users/yannbraga/open-source/storybook/scripts/task.ts:277:35)
                    at run (/Users/yannbraga/open-source/storybook/scripts/task.ts:411:34)
                    at processTicksAndRejections (node:internal/process/task_queues:96:5)
            </error>
        </testcase>
    </testsuite>
</testsuites>
```

image

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: scripts/tasks/build.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export const build: Task = {
     return pathExists(builtSandboxDir);
   },
   async run({ sandboxDir }, { dryRun, debug }) {
+    throw new Error('Problems!');
```
Comment (tmeasday) [inline scripts/tasks/build.ts:12]: No problems please!

**Review by tmeasday — APPROVED** [review]
OK!, seems like an improvement

File: scripts/tasks/build.ts (hunk 1 above)
Comment (yannbf) [reply scripts/tasks/build.ts:12]: 😂

Comment (yannbf) [reply scripts/tasks/build.ts:12]: This is how it looks like now

image

**Pull request merged**